educator/src/Dscp/Educator/Web/Educator/Types.hs

```diff
@@ -120,6 +122,66 @@ requestToAssignment NewAssignment{..} =
     , _aDesc = naDesc
     }
 
+---------------------------------------------------------------------------
+-- Buildable instances
+---------------------------------------------------------------------------
+
+instance Buildable (NewCourse) where
+    build (NewCourse{..}) =
+      "{ course id = " +| ncId |+
+      " }"
+
+instance Buildable (NewGrade) where
+    build (NewGrade{..}) =
+      "{ submission hash = " +| ngSubmissionHash |+
+      " }"
+
+instance Buildable (NewAssignment) where
+    build (NewAssignment{..}) =
+      "{ course id = " +| naCourseId |+
+      " }"
+
+instance Buildable (EnrollStudentToCourse) where
+    build (EnrollStudentToCourse{..}) =
+      "{ course id = " +| escCourseId |+
+      " }"
+
+instance Buildable (CourseEducatorInfo) where
+    build (CourseEducatorInfo{..}) =
+      "{ course id = " +| ciId |+
+      " }"
+
+instance Buildable (AssignmentEducatorInfo) where
+    build (AssignmentEducatorInfo{..}) =
+      "{ course id = " +| aiCourseId |+
+      " }"
+
+instance Buildable (SubmissionEducatorInfo) where
+    build (SubmissionEducatorInfo{..}) =
+      "{ hash = " +| siHash |+
+      " }"
+
+instance Buildable (ForResponseLog NewCourse) where
+    build = buildForResponse
+
+instance Buildable (ForResponseLog NewGrade) where
+    build = buildForResponse
+
+instance Buildable (ForResponseLog NewAssignment) where
+    build = buildForResponse
+
+instance Buildable (ForResponseLog EnrollStudentToCourse) where
+    build = buildForResponse
+
+instance Buildable (ForResponseLog CourseEducatorInfo) where
+    build = buildForResponse
+
+instance Buildable (ForResponseLog AssignmentEducatorInfo) where
+    build = buildForResponse
+
+instance Buildable (ForResponseLog SubmissionEducatorInfo) where
+    build = buildForResponse
```

Comment:
I think that you should've _only_ define `Buildable` instances for `ForResponseLog ...` wrapped types (like in `Dscp.Witness.Web.Types`) - the idea is that `Buildable` instance for type itself may be different than its `ForResponseLog` representation.